nixos/modules/services/search/meilisearch.nix

```diff
@@ -0,0 +1,99 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.services.meilisearch;
+
+in {
+
+  meta.maintainers = with maintainers; [ filalex77 ];
+
+  ###### interface
+
+  options.services.meilisearch = {
+    enable = mkEnableOption "MeiliSearch - a RESTful search API";
+
+    package = mkOption {
+      description = "MeiliSearch package to use.";
+      default = pkgs.meilisearch;
+      defaultText = "pkgs.meilisearch";
+      type = types.package;
+    };
+
+    listenAddress = mkOption {
+      description = "MeiliSearch listen address.";
+      default = "127.0.0.1";
+      type = types.str;
+    };
+
+    listenPort = mkOption {
+      description = "MeiliSearch port to listen on.";
+      default = 7700;
+      type = types.port;
+    };
+
+    dbPath = mkOption {
+      description = "Defines the location for the database files.";
+      default = "/var/lib/meilisearch";
+      type = types.path;
+    };
+
+    environment = mkOption {
+      description = "Defines the running environment of MeiliSearch.";
+      default = "development";
+      type = types.enum [ "development" "production" ];
+    };
+
+    masterKey = mkOption {
+      description = "Defines the default admin API key.";
+      default = null;
+      type = with types; nullOr str;
+    };
+
+    noAnalytics = mkOption {
+      description = ''
+        Deactivates analytics.
+        Analytics allow MeiliSearch to know how many users are using MeiliSearch,
+        which versions and which platforms are used.
+        This process is entirely anonymous.
+      '';
+      default = false;
+      type = types.bool;
+    };
+
+  };
+
+  ###### implementation
+
+  config = mkIf cfg.enable {
+    systemd.services.meilisearch = {
+      description = "MeiliSearch daemon";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      environment = {
+        MEILI_DB_PATH = cfg.dbPath;
+        MEILI_HTTP_ADDR = "${cfg.listenAddress}:${toString cfg.listenPort}";
+        MEILI_MASTER_KEY = mkIf (cfg.masterKey != null) (toString cfg.masterKey);
+        MEILI_NO_ANALYTICS = toString cfg.noAnalytics;
+        MEILI_ENV = cfg.environment;
+      };
+      serviceConfig = {
+        ExecStart = "${cfg.package}/bin/meilisearch";
+        User = "meilisearch";
+      };
+    };
+
+    environment.systemPackages = [ cfg.package ];
+
+    users = {
+      groups.meilisearch.gid = config.ids.gids.meilisearch;
+      users.meilisearch = {
+        uid = config.ids.uids.meilisearch;
+        description = "MeiliSearch daemon user";
+        home = cfg.dbPath;
+        createHome = true;
```

Comment:
Use either `serviceConfig.StateDirectory` or `systemd.tmpfiles.rules` to provision these types of directories.